### PR TITLE
Redefine responsibilities of WorkspaceServiceTest

### LINKED ISSUE #177: Redefine responsibilities of WorkspaceServiceTest
## Description ##
As a SE, I would like to adjust the unit test of the WorkspaceServiceTest to reflect the updated responsibility it have within the server.
The effort will be focus to remove unit test and activities that deals with the file-system that are now managed by the FileSystemService.
Using Mockito provide a mock version of WorkspaceService for testing purpose and address the related code smells defined below:
- [code smell reference list](https://sonarcloud.io/project/issues?id=eclipse_che-che4z-lsp-for-cobol&open=AXAMMvWiINNIWu5k3ITp&resolved=false)

The following classes may be affected:
- ```WorkspaceServiceTest```
- ```MockWorkspaceService```
- ```CobolWorkspaceServiceImpl```

**Subtasks status:**
- [X] Address code smells and get rid of CobolWorkspaceService 
- [X] Use Mockito to get rid of the Mock classes used during the test  (as MockWorkspaceService)
 
## Acceptance Criteria ##
- Code coverage is not affected and still met the quality gate
- no unit test are failing and the result are meaningful
- no more code smell are showed for the classes defined above

## Definition of Done ##
- The story is implemented according to the description.
- Unit testing done (Preferably by TDD approach, coverage should be at least 70%).
- Code review done by at least one other team member.
- Manual testing is done.
- Documentation is added and reviewed (if needed).
- The pull request is approved.
- The story is merged into the development branch.
- Tests are performed on the development branch.

## Technical details ##
This is the place for any implementation-related notes

### PR DESCRIPTION
**Pull request by zacanbrcom**
Purpose of this PR is to fix #177 with a planned activity to refactor the workspace unit test that doesn't need anymore to deal with copybook content (find, grab content and so on) that now are defined as responsibility of ```CopybookService```

### CODE REVIEW

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/broadcom/lsp/cdi/module/service/ServiceModule.java [hunk 1]
```diff
@@ -53,9 +54,10 @@ protected void configure() {
     bind(LanguageEngineFacade.class).to(CobolLanguageEngineFacade.class);
     bind(CopybookService.class).to(CopybookServiceImpl.class);
     bind(CopybookDependencyService.class).to(CopybookDependencyServiceImpl.class);
-    bind(CobolWorkspaceService.class).to(CobolWorkspaceServiceImpl.class);
+    bind(WorkspaceService.class).to(CobolWorkspaceServiceImpl.class);
```
Comment (sergiuilie) [inline com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/broadcom/lsp/cdi/module/service/ServiceModule.java]: binding appears two times into this file

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CobolWorkspaceServiceImpl.java [hunk 2]
```diff
@@ -50,16 +50,15 @@
...
+public class CobolWorkspaceServiceImpl implements WorkspaceService {
   private DataBusBroker dataBus;
   private Provider<LanguageClient> clientProvider;
 
   @Inject
   public CobolWorkspaceServiceImpl(DataBusBroker dataBus, Provider<LanguageClient> clientProvider) {
     this.dataBus = dataBus;
     this.clientProvider = clientProvider;
-    dataBus.subscribe(DataEventType.REQUIRED_COPYBOOK_EVENT, this);
+    dataBus.subscribe(DataEventType.REQUIRED_COPYBOOK_EVENT);
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CobolWorkspaceServiceImpl.java]: We do not need this subscription at all due to CobolWorkspaceServiceImpl does not accept any events

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestCopybookWithIndirectRecursiveDependencyIsDetected.java [hunk 3]
```diff
@@ -43,14 +45,25 @@
...
+
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(
+        () ->
+            Arrays.asList(
+                new CobolText("INNER-COPY", INNER_COPY),
+                new CobolText("INDIRECT-COPY", INDIRECT_COPY)));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestCopybookWithIndirectRecursiveDependencyIsDetected.java]: We don't need this code, the same is done at lines 63-66

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestCopybookWithRecursiveDependencyIsDetected.java [hunk 4]
```diff
@@ -43,6 +45,13 @@
 
   public TestCopybookWithRecursiveDependencyIsDetected() {
     super(TEXT);
+
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(
+        () -> Collections.singletonList(new CobolText("RECURSIVE-COPY", RECURSIVE_COPY)));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestCopybookWithRecursiveDependencyIsDetected.java]: Same here, we don't need this configuration

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestMissingCopybookNotInVariableList.java [hunk 5]
```diff
@@ -39,6 +44,14 @@
...
           + "       MOVE 00 TO CHILD1 OF PARENT.";
 
+  public TestMissingCopybookNotInVariableList() {
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(Collections::emptyList);
+  }
+
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestMissingCopybookNotInVariableList.java]: Same here

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestSameCopybooksWithDifferentCases.java [hunk 6]
```diff
@@ -44,8 +50,15 @@
...
-  public TestSameCopybooksWIthDifferentCases() {
+  public TestSameCopybooksWithDifferentCases() {
     super(TEXT);
+
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(
+        () -> Collections.singletonList(new CobolText("STRUCT1", STRUCT1)));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestSameCopybooksWithDifferentCases.java]: Same here

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestVariableStructureInCopybooksWithMissingLevels.java [hunk 7]
```diff
@@ -45,6 +49,12 @@
 
   public TestVariableStructureInCopybooksWithMissingLevels() {
     super(TEXT);
+
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(() -> Collections.singletonList(new CobolText("STRUCT", STRUCT)));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestVariableStructureInCopybooksWithMissingLevels.java]: Same here

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestVariableStructureIsBuiltWithCopybooks.java [hunk 8]
```diff
@@ -44,6 +50,12 @@
 
   public TestVariableStructureIsBuiltWithCopybooks() {
     super(TEXT);
+    DataBusBroker databus =
+        Guice.createInjector(new DatabusModule()).getInstance(DataBusBroker.class);
+
+    MockCopybookServiceImpl copybookService = new MockCopybookServiceImpl(databus);
+    copybookService.setCopybooks(
+        () -> Collections.singletonList(new CobolText("COPYBOOK-CONTENT", COPYBOOK_CONTENT)));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestVariableStructureIsBuiltWithCopybooks.java]: And here

**Comment by zacanbrcom** [comment]
we forgot again the changelog!! 😭

**Comment by temanbrcom** [comment]
Cool, thanks!

**Comment by zacanbrcom** [comment]
@shciubrcom @temanbrcom Sorry guys could u double check if the changelogs contains your contribution

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,9 +3,11 @@
 All notable changes to the COBOL Language Support extension are documented in this file.
 
 ## [0.x.x] - TBD
-- Dependency file should contain 8 char max copybook name [#175](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/175)
-- StatusBar copybook downloader profile switcher [#239](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/239)
-- Move fetch copybook quickfix from LSP server to extension. [#235](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/235)
+- Redefine responsibilities of WorkspaceServiceTest [232](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/232) Thanks @[zacanbrcom](https://github.com/zacanbrcom)
+- Remove copybooks Quick Fix on the server side due to it is a client responsibility [234](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/234) Thanks @[temanbrcom](https://github.com/temanbrcom)
```
Comment (temanbrcom) [inline CHANGELOG.md]: This change message is in a little conflict with one line 10. I think we may remove this one

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,9 +3,11 @@
...
+- Dependency file should contain 8 char max copybook name [#175](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/175) Thanks @[ishche](https://github.com/ishche)
+- StatusBar copybook downloader profile switcher [#240](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/240) Thanks @[ishche](https://github.com/ishche)
+- Move fetch copybook quickfix from LSP server to extension. [#235](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues/235) Thanks @[ishche](https://github.com/ishche)
```
Comment (temanbrcom) [inline CHANGELOG.md]: quickfix here is a typo

**Pull request merged**